semgrep/tests/acceptance/rule.yaml

```diff
@@ -0,0 +1,6 @@
+rules:
+  - id: useless-equals
+    pattern: $X==$X
+    message: "Useless equals"
+    languages: [python, javascript]
```

Comment:
You should add all our supported languages here [java, c, go]

Reply:
I think the canonical test for NaN in Go might be `$X != $X`.

Reply:
I think can go in a follow on PR. The given repos are all JS/Python for now.